dotnet/Capstone/Controllers/BirdSightingController.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Capstone.DAO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System;
+using Capstone.Models;
+using Capstone.Exceptions;
+using System.Security.Principal;
+
+namespace Capstone.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class BirdSightingController : ControllerBase
+    {
+        private readonly IBirdSightingDao birdSightingDao;
+        public BirdSightingController(IBirdSightingDao birdSightingDao)
+        {
+            this.birdSightingDao = birdSightingDao;
+
+        }
+
+        [HttpGet("/birds/{birdId}/notes")]
+        public List<BirdSighting> ListAllBirds(int birdId)
+        {
+            List<BirdSighting> sightingList = birdSightingDao.getSightings(birdId);
+
+            if (sightingList == null)
+            {
+                Console.WriteLine("No bird");
+            }
+
+            return sightingList;
+
+        }
+
+        [HttpPost("/newNote")]
+        public IActionResult addSighting( BirdSighting newBirdSighting)
+        {
+
+            IIdentity user = User.Identity;
+            
+
+            const string errorMessage = "An error occurred and a bird was not created.";
+
+            IActionResult result;
+            try
+            {
+                BirdSighting birdSighting = birdSightingDao.addSighting(newBirdSighting, user.Name);
+
+                result = Created("", newBirdSighting);
+            }
+            catch (DaoException e)
+            {
+                result = BadRequest(new { message = e.Message });
+            }
+            catch (ArgumentException e)
+            {
+                result = BadRequest(new { message = e.Message });
+            }
+            catch
+            {
+                result = BadRequest(new { message = errorMessage });
+            }
+
+            return result;
+
+
+        }
+        [HttpPut("/editNote/{id}")]
+        public IActionResult editSighting(BirdSighting sighting, int id)
+        {
+            
+            const string errorMessage = "An error occurred and bird could not be modified.";
+            IActionResult result;
+            try
+            {
+                birdSightingDao.editSighting(sighting, id);
+                result = Ok();
+            }
+            catch (DaoException e)
+            {
+                result = BadRequest(new { message = e.Message });
+            }
+            catch (ArgumentException e)
+            {
+                result = BadRequest(new { message = e.Message });
+            }
+            catch
+            {
+                result = BadRequest(new { message = errorMessage });
+            }
+
+
+            return result;
+        }
+         
+        [HttpDelete("/deleteNote/{id}")]
```

Comment:
You might limit this to deleting a note that the logged in user created

Reply:
I can add for the user to match but how will it even work in the front end? Will all the notes have a delete button next to them? Should I maybe create a new method where I am selecting Notes by user id that will have all the Notes created by a specific user and that will be the list of Notes that will have the delete button?

Reply:
I believe the notes/sightings will be added to Lists (as described in our user stories). We have a [user story](https://trello.com/c/Drmxv1iP) that indicates that we will be deleting from there, and it looks like the method is called in the `BirdNotes.vue` component.

Because the lists will be tied to the user, you shouldn't need to create a new method for the notes, I don't think? We can always circle back to that in another pull request if I'm wrong.